src/mongocxx/uri.cpp

```diff
@@ -26,6 +26,19 @@
 #include <mongocxx/config/private/prelude.hh>
 
 namespace mongocxx {
+
+namespace {
+
+std::string to_string_null_safe(const char* str)
+{
+    if (str == nullptr) {
+        return std::string();
+    }
+    return str;
+}
+
+}
```

Comment:
Stylistically, this is not the most elegant approach ever.  But it works.  I'm guessing that this function may be useful for other wrapper classes as well, but I didn't see a more central place to put it.